xmantissa/websession.py

```diff
@@ -241,20 +301,39 @@ def savorSessionCookie(self, request):
         """
         Make the session cookie last as long as the persistant session.
 
+        @type request: L{nevow.inevow.IRequest}
         @param request: The HTTP request object for the guard login URL.
         """
         cookieValue = request.getSession().uid
         request.addCookie(
-            self.cookieKey, cookieValue, path='/', max_age=PERSISTENT_SESSION_LIFETIME,
+            self.cookieKey, cookieValue, path='/',
+            max_age=PERSISTENT_SESSION_LIFETIME,
             domain=self.cookieDomainForRequest(request))
 
 
     def login(self, request, session, creds, segments):
-        """Called to check the credentials of a user.
+        """
+        Called to check the credentials of a user.
 
-        Here we extend guard's implementation to preauthenticate users
-        if they have a valid persistant session.
+        Here we extend guard's implementation to preauthenticate users if they
+        have a valid persistent session.
+
+        @type request: L{nevow.inevow.IRequest}
+        @param request: The HTTP request being handled.
+
+        @type session: L{nevow.guard.GuardSession}
+        @param session: The user's current session.
+
+        @type creds: L{twisted.cred.credentials.ICredentials}
+        @param creds: The credentials the user presented.
+
+        @type segments: L{tuple}
+        @param segments: The remaining segments of the URL.
+
+        @return: A deferred firing with the user's avatar.
         """
+        self._maybeCleanSessions()
```

Comment:
The case you mentioned is covered by `test_sessionCleanup`, did you mean something else?